### ISSUE TITLE
[bug] `conan config install` does not work on mounted network drives on Windows

### ISSUE THREAD

**Issue by yeputons** [conan-io/conan#14178]
### Environment details

* Operating System+version: Windows 10
* Compiler+version: any
* Conan version: 1.59.0
* Python version: 3.11.2

### Steps to reproduce

1. Run the following as Administrator in an elevated prompt to create a share:
```
mkdir C:\tmp-conan-share
net share tmp-conan-share=C:\tmp-conan-share /GRANT:EVERYONE,FULL
```
2. Run the following in your user's prompt to mount the share as a drive and try to run Conan on it. You can use arbitrary config.
```
net use U: \\127.0.0.1\tmp-conan-share
set CONAN_HOME=U:\.conan
set CONAN_USER_HOME=U:\.conan2
conan config install https://github.com/conan-io/conanclientcert.git
```
3. Observe that Conan fails very early on, in a way very similar to one described in #6173. However, here it involves real network drives, not just a simple `subst` call, so it still does not work.
4. Unmount the share:
```
 net use U: /DELETE 
```
5. Remove the share by running the following commands in an elevated command line:
```
net share tmp-conan-share /delete
del /F C:\tmp-conan-share
```

I believe the underlying reason is that Python's `subprocess.Popen` does not work on network shares with `shell=True` because `cmd.exe` itself does not support it. Moreover, mounting network share as a drive does not help because Conan decides to resolve the path for the temporary directory and ends up on the network share nonetheless: https://github.com/conan-io/conan/blob/b43eb83956f053a47cc3897cfdd57b9da13a16e6/conans/client/conf/config_installer.py#L50

If I comment out this line, at least `conan config install` completes successfully, but I wouldn't be surprised if there are similar troubles in other places e.g. in caches or build operations.

### Logs

```
C:\Users\egor.suvorov>conan config install https://github.com/conan-io/conanclientcert.git
Trying to clone repo: https://github.com/conan-io/conanclientcert.git
ERROR: Failed conan config install: Can't clone repo: '\\127.0.0.1\tmp-conan-share\.conan\tmp_config_install'
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
fatal: destination path '.' already exists and is not an empty directory.
```

**Comment by memsharded** [comment]
Hi @yeputons 

Thanks for  your report.
I'd like to highlight some important issues, before proceeding to check the report.

- The Conan cache is designed as a local, per-user, per-machine cache, not designed to be shared among multiple users or multiple machines
- Conan 1.X cache has some locking (partial, not complete, not guaranteed) capabilities for some limited concurrency support, but those are OS resources, so it will  not work across different machines. That is, the cache is not not multi-machine concurrent, and if any 2 machines use it concurrently, it will break and left in unusable state.
- Conan 2.0 uses more extensively a sqlite DB, which access might also be not concurrent-safe for multi-machine.

So, my recommendation would be, basically don't. This can have too many problems, and it is not worth those problems, just to try to save some disk space, if that is the reason for doing this.

Regarding your report, I think this can be solved by `` tmp_folder = os.path.realpath(tmp_folder)`` => `` tmp_folder = os.path.abspath(tmp_folder)``, I am fine changing it if that helps, but I'd strongly recommend re-considering the approach as commented above.

**Comment by yeputons** [comment]
@memsharded Thank you. In my particular situation the network share is not shared between machines, so it seems I'm safe. The share is purely to provide more disk space for one specific user as local drive on that particular VM is tiny. The reason for such setup is, uhm, mostly bureaucratic.

**Comment by memsharded** [comment]
Sounds good. Do you want to try to modify that ``realpath``=>``abspath`` and see if it solves it in your setup? That would really help (doing this kind of testing in automatic tests and in CI is quite complicated)

**Comment by yeputons** [comment]
@memsharded Sure. It may take some weeks, though, as I may have some project-specific bureaucracy issues as well, it's the first time I'm trying to run the project in that environment. I'll let you know here, just ping me if I forget.

**Comment by yeputons** [comment]
@memsharded It seems like configuring the project worked for me with your change from `realpath` to `abspath`. I will get back once I figure out my dependencies and build the project.

I've also added `assert not os.getcwd().startswith('\\\\')` next to all places that use `shell=True`, none of them failed. Trouble is, running within UNC path does not result in an immediate error, it just changes the current directory to `C:\Windows`.

**Comment by memsharded** [comment]
I have also done #14183 because there is a concerning comment there saying that the ``realpath`` might be necessary for OSX users (introduced in https://github.com/conan-io/conan/pull/4709/files), I'll try to follow up on that potential issue too.

**Comment by yeputons** [comment]
@memsharded Happy to let you know that the project built without any troubles with the `abspath` -> `realpath` workaround, only using the network drive.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14183 was merged for 2.0.10, which should also avoid issues with this. Feedback welcome!
